ilmoituslomake/opening_times/views.py

```diff
@@ -0,0 +1,138 @@
+from django.shortcuts import get_object_or_404
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.generics import RetrieveAPIView, UpdateAPIView
+import requests
+from datetime import datetime, timedelta
+from notification_form.models import Notification
+from opening_times.utils import (
+    create_hauki_resource,
+    create_url,
+    partially_update_hauki_resource,
+    update_origin,
+)
+from ilmoituslomake.settings import HAUKI_API_URL
+
+# Permissions
+from rest_framework.permissions import IsAuthenticated
+
+
+class CreateLink(UpdateAPIView):
+    permission_classes = [IsAuthenticated]
+    # permission_classes = (permissions.AllowAny,)
+
+    def post(self, request, id=None, *args, **kwargs):
+
+        # Request params
+        request_params = request.data
+        name = request_params["name"]
+        description = request_params["description"]
+        address = request_params["address"]
+        resource_type = request_params["resource_type"]
+        is_public = True
+        timezone = request_params["timezone"]
+
+        # id and hauki_id must be string
+        hauki_id = str(request_params["hauki_id"])
+        id = str(id)
+
+        # hsa_resource is hauki_id if "kaupunkialusta:id" does not exist in hauki.
+        hsa_resource = hauki_id
+
+        # Check if resource exists
+        response = requests.get(HAUKI_API_URL + "kaupunkialusta:" + id + "/")
```

Comment:
Add some small but reasonable timeout value here. We don't want to wait too long here.